packages/nextjs/components/scaffold-eth/Faucet.tsx

```diff
@@ -74,7 +77,7 @@ export const Faucet = () => {
   };
 
   // Render only on local chain
-  if (!ConnectedChain || ConnectedChain.id !== hardhat.id) {
+  if (ConnectedChain?.id !== hardhat.id || !isMounted()) {
```

Comment:
Don't think we need `isMounted` here. This line is only evaluated during rendering of this component, at which stage `isMounted()` should always return true. When the component isn't mounted, this code will not be executed.

(Checking `isMounted` is more for async stuff, for example when waiting for a response from an API, it's possible the user navigates away and the component that made the request gets unmounted.)

Reply:
`isMounted` is `false` by default and becomes `true` in `useEffect` after the first render. I added it because it fixes hydration error - `useNetwork` works differently for nextjs server and client

Reply:
Yeah! I think it's because of NextJS pre-rendering. the `isMounted` logic makes that pre-rendering === first render. If not, we'd get a Hydration error.

Reply:
Aaah of course, should have realized in the context of this fix, you're right @rin-st